src/Validation/JpValidation.php

```diff
@@ -25,13 +25,46 @@ class JpValidation extends LocalizedValidation
 {
     /**
      * Checks a phone number for Japan.
+     * Accepts the following format.
+     * - 0311112222
+     * - 03 1111 2222 or 09876 1 2222
+     * - 03-1111-2222 or 09876-1-2222
+     * - +81311112222
+     * - +81 3 1111 2222
+     * - +81-3-1111-2222
+     * - 09011112222
+     * - 090 1111 2222
+     * - 090-1111-2222
+     * - +819011112222
+     * - +81 90 1111 2222
+     * - +81-90-1111-2222
+     * - 0120111222
+     * - 0120 111 222
+     * - 0120-111-222
      *
      * @param string $check The value to check.
      * @return bool Success.
      */
     public static function phone($check)
     {
-        $pattern = '/^(0\d{1,4}[\s-]?\d{1,4}[\s-]?\d{1,4}|\+\d{1,3}[\s-]?\d{1,4}[\s-]?\d{1,4}[\s-]?\d{1,4})$/';
+        $pattern = '/^(?:(?:0|\+81[\s\-]?)[1-9](?:0(?:\d{8}|(?:[\s\-]\d{4}){2})|(?:[1-9]\d{7}|(?:[1-9]0(?:[\s\-]\d{3}){2}|\d{0,3}[\s\-]\d{1,4}[\s\-]\d{4}))))$/';
```

Comment:
@slywalker @dereuromark Also, what do you think about adding the `D` modifier to the regular expression? Currently, it matches also a LF if it is at the end of the string.